09-Time-Series-Data.ipynb

```diff
@@ -0,0 +1,786 @@
+    "<br/>\n",
+    "<section class=\"objectives panel panel-success\">\n",
+    "<div class=\"panel-heading\">\n",
+    "<h2><span class=\"fa fa-pencil\"></span> Google your troubles away </h2>\n",
+    "</div>\n",
+    "<br/>\n",
+    "So the docs are [here](http://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.genfromtxt.html). Find the syntax to change to names to better to represent maximum length, lifetime and point in time which they occured.\n",
+    "<br/>\n",
+    "</section>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "data.dtype.names = ('max_len', 'ltime', 'sample_time')\n",
+    "data['max_len']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### DataFrame\n",
+    "\n",
+    "Now a pandas dataframe takes two arguments as a minimum, index and data. In this case the index will be our time within the sample and the maximum length and lifetime will be our data. So lets import pandas and use the dataframe:\n",
+    "\n",
+    "Now pandas reads a dictionary when we want to input multiple data columns. Therefore we need to make a dictionary of our data and read that into a pandas data frame. First however though we need to import pandas.\n",
+    "\n",
+    "<section class=\"objectives panel panel-info\">\n",
+    "<div class=\"panel-heading\">\n",
+    "<h2><span class=\"fa fa-thumb-tack\"></span> Dictionaries </h2>\n",
+    "</div>\n",
+    "<br/>\n",
+    "So we covered dictionaries earlier. We can create keyword data pairs to form a dictionary (shock horror) of values. In this case \n",
+    "\n",
+    "<pre>\n",
+    "temps = {'Brussles': 9, 'London': 3, 'Barcelona': 13, 'Rome': 16}\n",
+    "temps['Rome']\n",
+    "16\n",
+    "</pre>\n",
+    "\n",
+    "We can also find out what keywords are associated with a given dictionary, In this case:\n",
+    "\n",
+    "<pre>\n",
+    "temps.keys()\n",
+    "dict_keys(['London', 'Barcelona', 'Rome', 'Brussles'])\n",
+    "</pre>\n",
+    "\n",
+    "</code>\n",
+    "<br/>\n",
+    "</section>\n",
+    "\n",
+    "First, let's import Pandas:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "d = {'max_len': data['max_len'], 'ltime': data['ltime']}\n",
+    "\n",
+    "df = pd.DataFrame(data=d, index=data['sample_time'])\n",
+    "print(df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Datetime Objects\n",
+    "\n",
+    "Notice that the time for the sample is in a strange format. It is a string containing the date in YYYY-MM-DD and time in HH-MM-SS-mmmmmm. These datetime objects have their own set of methods associated with them. Python appreciates that theses are built this way and can use them for the indexing easily. \n",
```

Comment:
Be sure to make it clear that datetime objects are not strings - they are their own distinct object that just prints as a string.